rotkehlchen/tests/api/test_yearn_vaults.py

```diff
@@ -82,9 +95,51 @@ def test_query_yearn_vault_v2_balances(rotkehlchen_api_server, ethereum_accounts
         else:
             result = assert_proper_sync_response_with_result(response)
 
-    for vault in result[TEST_V2_ACC2].values():
-        assert '%' in vault['roi']
-        assert FVal(vault['vault_value']['amount']) > ZERO
-        assert FVal(vault['vault_value']['usd_value']) > ZERO
-        assert FVal(vault['underlying_value']['amount']) > ZERO
-        assert FVal(vault['underlying_value']['usd_value']) > ZERO
+    assert_proper_vault_response(result[TEST_V2_ACC2])
+
+
+@pytest.mark.vcr(filter_query_parameters=['apikey'])
+@pytest.mark.parametrize('ethereum_accounts', [[TEST_V3_ACC3]])
+@pytest.mark.parametrize('ethereum_modules', [['yearn_vaults_v3']])
+@pytest.mark.parametrize('should_mock_current_price_queries', [True])
+@pytest.mark.parametrize('should_mock_price_queries', [True])
+@pytest.mark.parametrize('default_mock_price_value', [ONE])
+def test_query_yearn_vault_v3_balances(
+        rotkehlchen_api_server: 'APIServer',
+        ethereum_accounts: list['ChecksumEvmAddress'],
+):
```

Comment:
this test is not needed but you need to check that v3 tokens are detected properly and that we query their price. We have `find_yearn_price` that now handles v2 logic. Since v3 is similar to v2 probably you can reuse them or abstract most of it

Reply:
The `test_yearn_api` is testing that v3 tokens are retrieved. It had a v3 vault in its mocked api response already, so I just changed the assert to allow for that token to be detected instead of skipped (https://github.com/rotki/rotki/pull/8677/files#diff-c96dc6e72378cc3637aa60704469d066db0dc13b8edee57b724b6447cbac93a3R71).

I also added `test_find_yearn_vaults_v3_price` to check for querying the price.